app/src/main/java/org/thoughtcrime/securesms/repository/ConversationRepository.kt

```diff
@@ -180,59 +240,83 @@ class DefaultConversationRepository @Inject constructor(
         storage.setRecipientApproved(recipient, isApproved)
     }
 
-    override suspend fun deleteForEveryone(
+    override suspend fun deleteCommunityMessagesRemotely(
         threadId: Long,
-        recipient: Recipient,
-        message: MessageRecord
-    ): Result<Unit> = suspendCoroutine { continuation ->
-        buildUnsendRequest(recipient, message)?.let { unsendRequest ->
-            MessageSender.send(unsendRequest, recipient.address)
+        messages: Set<MessageRecord>
+    ) {
+        val community = lokiThreadDb.getOpenGroupChat(threadId) ?:
+            throw Exception("Not a Community")
+
+        messages.forEach { message ->
+            lokiMessageDb.getServerID(message.id, !message.isMms)?.let { messageServerID ->
+                OpenGroupApi.deleteMessage(messageServerID, community.room, community.server).await()
+            }
         }
+    }
+
+    override suspend fun delete1on1MessagesRemotely(
```

Comment:
I don't see anything that requires `suspend` in the function body?

Reply:
I made it a suspend as a safety mechanism to force the calling point to use this from a coroutine.
I guess another option would be to use a context directly in the method here to keep the responsibility local.